File: 4rth_Sem_c++_backup/IMP/Computational Lab/Assignment 1/trapezoidal.cpp
/*q7as1.cpp: Write a program to implement Trapezoidal Rule to calculate value of an integral
*/
#include <iostream>
using namespace std;
class Trapezoidal
{
  public:
	double trapezoidal(double a, double b, long int n);
	double function_1(double x);
};
double Trapezoidal :: trapezoidal(double a, double b, long int n)
{
	double yon, y1, it;
	long int i;
	double h;
	h = (b - a) / n;
	yon = function_1(a) + function_1(b);
	y1 = 0;
	for(i = 1; i < n; i++)
		y1 = y1 + function_1(a + i * h);
	it = .5 * h * (yon + 2 * y1);
	return it;
}
double Trapezoidal :: function_1(double x)
{
	double y;
	y = 4.0 / (1 + x * x);
	return y;
}
int main()
{
	double a, b, it;
	long int n;
	system("clear");
	cout << endl << "Enter Number of Intervals:  ";
	cin >> n;
	cout << "Enter Lower limit of your integral: ";
	cin >> a;
	cout << "Enter upper limit of your Integral: ";
	cin >> b;
	Trapezoidal T;
	it = T.trapezoidal(a, b, n);
	cout << "\nValue of Integral using Trapezoidal Rule: " << it << endl;
	return 0;
}

File: 5th_sem_C_Java_VB/Java/Numerical_Analysis/trapezoidal.java
import java.io.*;
class trapezoidal
{
  public double function_1(double x)
  {
    double y;
    y=4/(1+x*x);
    return y;
  }
  public double trap(double a,double b,int n)
  {
    double s1,s2,it,h;
    int i;
    double y[] = new double[1000];
    h=(b-a)/n;
    s1=function_1(a)+function_1(b);
    for(i=1;i<n;i++)
      y[i]=function_1(a+i*h);
    s2=0;
    for(i=1;i<n;i++)
      s2=s2+y[i];
    it=0.5*h*(s1+2*s2);
    return it;
  }
  public static void main(String args[])throws IOException
  {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    double a,b,it;
    int n;
    System.out.print("Enter lower limit of your integral : ");
    a=Double.parseDouble(br.readLine());
    System.out.print("Enter upper limit of your integral : ");
    b=Double.parseDouble(br.readLine());
    System.out.print("Enter number of intervals : ");
    n=Integer.parseInt(br.readLine());
    trapezoidal tr = new trapezoidal();
    it=tr.trap(a,b,n);
    System.out.println("Value of integral : "+it);
  }
}